src/main/java/com/cburch/logisim/fpga/gui/FPGACommander.java

```diff
@@ -223,11 +211,10 @@ private JPanel getExecuteWindow() {
     JPanel pan1 = new JPanel();
     pan1.setLayout(new BorderLayout());
     pan1.add(textMainCircuit, BorderLayout.WEST);
-    circuitsList.setActionCommand("mainCircuit");
     RebuildCircuitSelection();
     MyProject.addProjectListener(this);
     MyProject.getLogisimFile().addLibraryListener(this);
-    circuitsList.setActionCommand("Circuit");
+    circuitsList.setActionCommand("mainCircuit");
```

Comment:
can we have the `"mainCircuit"` string a final static?

Reply:
There are a lot of things in this file that needs refactoring, leaving it for the moment as is.